src/main/java/mods/eln/node/NodeBase.java

```diff
@@ -153,6 +153,10 @@ public NodeBase() {
 
     boolean destructed = false;
```

Comment:
> It'll work, no doubt about that, but only so long as a bunch of special cases are caught. What's actually going on is... the node is getting reconnected after it's been supposedly destroyed, when the UI closes. A symptom of the simulation existing in a parallel world from Minecraft, I suppose; the block is gone, but the node is not.

The parallel existence is not optional. It is required as nodes which's blocks are not actually loaded have to be simulated too. The *destructed* attribute was already present before, imho this is the least intrusive way to fix the issue by just using it as how it was actually intended.

> Ideally we'd tie them a bit more closely together. How often is connectJob called? Often enough... checking for the block's existence every time it's run might degrade performance, or have other problems.

This method is only called when there is an interaction of the user with the block or one of it's neighbors or a block gets destroyed. I would not call this often. Further the performance impact of reading this boolean should be not significant at all.

I can remove the check in the connectJob() method, it is just a second test that might catch (or not) other issues, the inventory issue should be already fixed in the respective Inventory classes. But for the actual costs I think it is worth to keep it.

> Maybe we could add a process that, at a relatively slow pace, checks each node to make sure it still exists in-world? That would also catch other causes of this sort of thing. It should cry havoc to the console if it finds anything, of course, but it's a fallback and debugging mechanism.

Imho maybe additionally having such a test in debug mode would make sense to find problems, but I think the impact to the performance could be significant, as this check might force Minecraft to load offline chunks just to do the check.